### PR TITLE
Add proxy_config field to containerattached resource.

### PR DESCRIPTION
**Pull request by loanchip**
Adds proxy support to `containerattached` resource.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
containerattached: added `proxy_config` field to `google_container_attached_cluster` resource
```

### CODE REVIEW

File: mmv1/products/containerattached/Cluster.yaml
Comment (roaks3) [inline mmv1/products/containerattached/Cluster.yaml:341]: Can any of these values be changed without recreating the whole cluster? If so, the update tests should probably be changed to alter the values and verify the update works. If not, then some of these fields should probably have `immutable: true` so that they instruct Terraform to recreate the cluster.

File: mmv1/products/containerattached/Cluster.yaml
Comment (roaks3) [inline mmv1/products/containerattached/Cluster.yaml:342]: Nit: `false` is the default, so this line can be removed

File: mmv1/products/containerattached/Cluster.yaml
Comment (roaks3) [inline mmv1/products/containerattached/Cluster.yaml:345]: For `proxyConfig` and `kubernetesSecret`, you may want to consider if you need `allow_empty_object: true`. It may not make sense to explicitly specify either of them as empty config objects, unless perhaps that is useful for clearing values. I see it on most of the other fields in this resource, so maybe easiest to just stay consistent.

**Review by roaks3 — CHANGES_REQUESTED** [review]
Changes look good to me, just a few comments on how the field is configured. Main blocker is the update behavior: it should be immutable, or test that update-in-place is working.

**Comment by roaks3** [comment]
I don't believe this will take much more after addressing my comments, but I will be OOO for the rest of the week, so transferring this to another reviewer in case this moves forward during that time.

File: mmv1/products/containerattached/Cluster.yaml
Comment (loanchip) [reply mmv1/products/containerattached/Cluster.yaml]: Removed, thanks.

File: mmv1/products/containerattached/Cluster.yaml
Comment (loanchip) [reply mmv1/products/containerattached/Cluster.yaml:341]: Good point. We do allow updating in place wothout having to recreate the cluster.
Updated the test to have this scenario.

File: mmv1/products/containerattached/Cluster.yaml
Comment (loanchip) [reply mmv1/products/containerattached/Cluster.yaml]: Yes, removed this since we don't consume empty objects for updates.

**Review by shuyama1 — APPROVED** [review]
LGTM! Thank you!

**Comment by shuyama1** [comment]
Looks like the PR may contain some old VCR cassettes, therefore triggering a rerun just in case. I'll merge it after the tests are done.
/gcbrun

**Review by shuyama1 — APPROVED** [review]

**Comment by shuyama1** [comment]
Sorry, I'm sure why the merging is still blocked. It may need approval from @roaks3 since they requested changes earlier. Pass it to @roaks3 then.

**Review by roaks3 — APPROVED** [review]
Ah yea, I was afraid of that. I'm not sure there is a way to clear a previous rejection :/

**Pull request merged**
